### PR TITLE
Feature offline mode, run local models when offline

### LINKED ISSUE #281: local mode doesn't work without internet
### Describe the bug

interpreter throws error regarding network connection failure when running in local mode disconnected from the internet. it throws errors twice. first, even before asking for model selection. second, after selecting the model.

ps: attached all errors as screenshots below.

> note that i do have the model locally installed already!

### Reproduce

1. Install open-intrepreter
2. Run it on local mode for the first time when connected to internet
3. Download the required Code Llama model following the instructions
4. Exit interpreter
5. Disconnect from internet
6. Run it on local mode again selecting the same model

### Expected behavior

Work flawlessly without any errors on local mode even when disconnected from internet.

### Screenshots

image
image
image

### Open Interpreter version

0.1.2

### Python version

3.10.6

### Operating System name and version

macOS Ventura 13.5

### Additional context

I was not using a virtual environment throughout. (guessing that shouldn't be the problem?)

### PR DESCRIPTION
**Pull request by mrdiamonddirt**
### Describe the changes you have made:
Added a new command line arguement for if you want to run in offline mode --offline, made changes to the loading and selecting of local models to allow for loading a local model when there is no internet connection found in the open interpreter models folder

here is a gif of the functionality only able to test on windows
interpreter_21

### Reference any relevant issue (Fixes #000)
fixes #281

- [x] I have performed a self-review of my code:

### I have tested the code on the following OS:
- [x] Windows
- [ ] MacOS
- [ ] Linux

### AI Language Model (if applicable)
- [ ] GPT4
- [ ] GPT3
- [x] Llama 7B
- [x] Llama 13B
- [x] Llama 34B
- [x] Huggingface model (maybe all predownloaded offline ones)

### CODE REVIEW

**Review by acbp — APPROVED** [review]

File: interpreter/cli.py [hunk 1]
```diff
@@ -148,7 +155,22 @@ def cli(interpreter):
     # This way, when folks hit interpreter --local, they get the same experience as before.
     
     rprint('', Markdown("**Open Interpreter** will use `Code Llama` for local execution. Use your arrow keys to set up the model."), '')
-        
+    if interpreter.offline:
+      models = {}
+      # get all models in the models folder
+      user_data_dir = appdirs.user_data_dir("Open Interpreter")
```
Comment (acbp) [inline interpreter/cli.py:161]: doenst import appdirs

Comment (mrdiamonddirt) [reply interpreter/cli.py:161]: good catch, I think this pr maybe massively out of date now and may be in need of a complete rework to be intergrated into the current code base, I still think there is something to be said for ensuring open-interpreter can still run when there is no internet connection

**Comment by MikeBirdTech** [comment]
Hey @mrdiamonddirt 
Thanks for the time and effort that you put into this PR. Sorry that it fell through the cracks. Since OI has offline mode, are you ok if we close this? Thanks!

**Comment by mrdiamonddirt** [comment]
Absolutely Mike go for it

**Pull request closed without merging**